armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -128,9 +129,14 @@ def construct(
             if isinstance(c, Component):
                 # there are other things like composite groups that don't get
                 # material modifications -- skip those
-                validMatModOptions = signature(
-                    c.material.applyInputParams
-                ).parameters.keys()
+                validMatModOptions = set()
+                for materialParentClass in c.material.__class__.__mro__:
+                    # we must loop over parents as well, since applyInputParams
+                    # could call to Parent.applyInputParams()
+                    if issubclass(materialParentClass, Material):
+                        validMatModOptions |= signature(
```

Comment:
This might be a little easier to read as `validMatModOptions.update()` than `validMatModOptions |=`. 

It's an opinion, of course.